Problem Statement 1:
In each of the following situations, state whether it is a correctly stated hypothesis
testing problem and why?
1. H0: μ = 25, H1: μ ≠ 25 - Correct
2. H0: σ > 10, H1: σ = 10 - Non Correct
3. H0: x = 50, H1: x ≠ 50 - Correct
4. H0: p = 0.1, H1: p = 0.5 - Non Correct
5. H0: s = 30, H1: s > 30 - Correct

Problem Statement 2:
The college bookstore tells prospective students that the average cost of its textbooks is Rs. 52 with a standard deviation of Rs. 4.50. A group of smart statistics students thinks that the average cost is higher. To test the bookstore’s claim against their alternative, the students will select a random sample of size 100. Assume that the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the 5% level of significance and state your decision.


In [186]:
import numpy as np
import pandas as pd
from scipy import stats
import matplotlib.pyplot as plt
import math
from scipy.stats import chi2

In [58]:
mu=52
std=4.5
n=100
sa=52.80
alpha=0.05
alpha
# The sample size is more than 30 and based on central limit theorefore distribution of sample is normal.
#Z-test is used in this hypothesis test.

In [87]:
z_test = (sa-mu)/(std/(n**0.5))
print(z_test)
p = 1 - stats.norm.cdf(z_test,0,1)
print(p)
z = stats.norm.ppf(0.025,0,1)
print(z)

-2.023137479931484
0.9784705123462794
-1.9599639845400545


Ho= Average cost of book is 52
H1 = Average cost of book is not 52
Its two tail Z-test to large sample size and text book cost can be more than 52 or less than 52

In [92]:
if p < 0.025:
    print(f"""P-value ({p:1.3f}) is lower than significance level ({alpha/2:1.2f}), then Null hypothesis is rejected""")
else:
    print(f"""p-value ({p:1.4f}) is greater than significance lever {alpha/2:1.2f}. Therefore, null hypothesis is accepted, therfore average cost of book is 52.""")

p-value (0.9785) is greater than significance lever 0.03. Therefore, null hypothesis is accepted, therfore average cost of book is 52.


Problem Statement 3:
A certain chemical pollutant in the Genesee River has been constant for several years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A group of factory representatives whose companies discharge liquids into the river is
now claiming that they have lowered the average with improved filtration devices. A group of environmentalists will test to see if this is true at the 1% level of significance. Assume \ that their sample of size 50 gives a mean of 32.5 ppm.
Perform a hypothesis test at the 1% level of significance and state your decision.

In [99]:
mu=34
std=8
n=50
sa=32.5
alpha=0.01
critical=alpha/2
# The sample size is more than 30 and based on central limit theorefore distribution of sample is normal.
#Z-test is used in this hypothesis test.

In [100]:
z_test = (sa-mu)/(std/(n**0.5))
print(z_test)
p = 1 - stats.norm.cdf(z_test)
print(p)
z = stats.norm.ppf(0.005,0,1)
print(z)

-1.3258252147247767
0.9075512005172
-2.575829303548901


Ho = μ < 34 ppm 
Ha:  µ ≠  34ppm 
Z-test is used in this hypothesis test. Its two tail test. 

In [102]:
if p < 0.005:
    print(f"""P-value ({p:1.3f}) is lower than significance level ({critical:1.2f}), then Null hypothesis is rejected""")
else:
    print(f"""p-value ({p:1.4f}) is greater than significance level {critical:1.2f}. Therefore, null hypothesis is accepted, Mean is leass than 34 ppm.""")

p-value (0.9076) is greater than significance level 0.01. Therefore, null hypothesis is accepted, Mean is leass than 34 ppm.


Problem Statement 4:
Based on population figures and other general information on the U.S. population, suppose it has been estimated that, on average, a family of four in the U.S. spends about $1135 annually on dental expenditures. Suppose further that a regional dental
association wants to test to determine if this figure is accurate for their area of country. To test this, 22 families of 4 are randomly selected from the population in that area of the country and a log is kept of the family’s dental expenditure for one
year. The resulting data are given below. Assuming, that dental expenditure is normally distributed in the population, use the data and an alpha of 0.5 to test the dental association’s hypothesis.
1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994

In [11]:
mu=1135
n=22
df=22-1
alpha=0.05
critical=0.025
x=[1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994]
df_x=pd.DataFrame(x,columns= ['expense'])
sa = df_x['expense'].mean()
std= df_x['expense'].std()
# The sample size is more than 22 and based on central limit theorefore distribution of sample is normal.
#T-test is used in this hypothesis test.

In [12]:
T_test = (sa-mu)/(std/(n**0.5))
print(T_test)
p = 1 - stats.t.cdf(T_test,df=21)
print(p)
t = stats.t.ppf(0.025,21)
print(t)

-2.023137479931484
0.9720130840267708
-2.0796138447276626


Ho = μ = $1135 
Ha: µ ≠ $1135
Its two tail test.


In [15]:
if p < 0.025:
    print(f"""P-value ({p:1.3f}) is lower than significance level ({critical:1.2f}), then Null hypothesis is rejected""")
else:
    print(f"""p-value ({p:1.4f}) is greater than significance level {critical:1.2f}. Therefore, null hypothesis is accepted, average income of family is $1135.""")

p-value (0.9720) is greater than significance level 0.03. Therefore, null hypothesis is accepted, average income of family is $1135.


Problem Statement 5:
In a report prepared by the Economic Research Department of a major bank the Department manager maintains that the average annual family income on Metropolis is $48,432. What do you conclude about the validity of the report if a random sample
of 400 families shows and average income of $48,574 with a standard deviation of 2000?

In [30]:
mu=48432
n=400
sa=48574
std=2000
alpha=0.05
critical=alpha/2
# The sample size is more than 400 and based on central limit theorefore distribution of sample is normal.
#Z-test is used in this hypothesis test.

Ho = μ = $48432 
Ha: µ > $48432
Its two tail test.

In [31]:
z_test = (sa-mu)/(std/(n**0.5))
print(z_test)
p = 1 - stats.norm.cdf(z_test)
print(p)
z = stats.norm.ppf(0.05,0,1)
print(z)

1.42
0.07780384052654643
-1.6448536269514729


In [32]:
if p < 0.05:
    print(f"""P-value ({p:1.3f}) is lower than significance level ({critical:1.2f}), then Null hypothesis is rejected""")
else:
    print(f"""p-value ({p:1.4f}) is greater than significance level {critical:1.2f}. Therefore, null hypothesis is accepted, therefore average income is $48432.""")

p-value (0.0778) is greater than significance level 0.03. Therefore, null hypothesis is accepted, therefore average income is $48432.


Problem Statement 6:
Suppose that in past years the average price per square foot for warehouses in the United States has been $32.28. A national real estate investor wants to determine whether that figure has changed now. The investor hires a researcher who randomly
samples 19 warehouses that are for sale across the United States and finds that the mean price per square foot is $31.67, with a standard deviation of $1.29. assume that the prices of warehouse footage are normally distributed in population. If the
researcher uses a 5% level of significance, what statistical conclusion can be reached? What are the hypotheses?

In [40]:
mu=32.28
n=19
sa=31.67
std=1.29
alpha=0.05
critical=alpha/2
df=n-1
# The sample size is less than 19.
#T-test is used in this hypothesis test.

Ho = μ = $32.28 #Average price per square foot.
Ha: µ ≠ $32.28
Its two tail test.

In [43]:
T_test = (sa-mu)/(std/(n**0.5))
print(T_test)
p = 1 - stats.t.cdf(T_test,df=18)
print(p)
t = stats.t.ppf(0.025,18)
print(t)

-2.06118477175179
0.972983235420014
-2.10092204024096


In [44]:
if p < 0.025:
    print(f"""P-value ({p:1.3f}) is lower than significance level ({critical:1.2f}), then Null hypothesis is rejected""")
else:
    print(f"""p-value ({p:1.4f}) is greater than significance level {critical:1.2f}. Therefore, null hypothesis is accepted, average price per square foot is $32.28.""")

p-value (0.9730) is greater than significance level 0.03. Therefore, null hypothesis is accepted, average price per square foot is $32.28.


Problem Statement 8:
Find the t-score for a sample size of 16 taken from a population with mean 10 when the sample mean is 12 and the sample standard deviation is 1.5.

In [ ]:
mu=10
n=16
sa=12
std=1.5
df=n-1


In [47]:
T_score = (sa-mu)/(std/(n**0.5))
print(T_score)


-2.06118477175179


Problem Statement 9:
Find the t-score below which we can expect 99% of sample means will fall if samples of size 16 are taken from a normally distributed population.

In [8]:
n=16
df=n-1
alpha=0.01
t = stats.t.ppf(0.01,15)
print(t)


-2.6024802949954933


Problem Statement 10:
If a random sample of size 25 drawn from a normal population gives a mean of 60 and a standard deviation of 4, find the range of t-scores where we can expect to find the middle 95% of all sample means. Compute the probability that (−t0.05 <t<t0.10).

In [12]:
mu = 60
n=25
std=4
alpha=0.05


In [16]:
t = stats.t.ppf(0.05,24)
print(t)
range1 = mu + t_value*(std/(n**0.5))
range2 = mu - t_value*(std/(n**0.5))
print(f"95% CI: {range1}, {range2}")

-1.7108820799094282
95% CI: 60.40789533106342, 59.59210466893658


Problem Statement 11:
Two-tailed test for difference between two population means Is there evidence to conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week, given the following:
Population 1: Bangalore to Chennai n1 = 1200
x1 = 452
s1 = 212
Population 2: Bangalore to Hosur n2 = 800
x2 = 523
s2 = 185

In [19]:
n1=1200
x1=452
s1=212

n2=800
x2=523
s2=185

Ho = There is no change two population mean i e x1=x2
Ho = There is difference between two population mean x1=!x2

In [30]:
t_value=(x1-x2)/((s1/(n1**0.5)+(s2)/(n2**0.5)))
print(t_value)
p = 1 - stats.t.cdf(t_value,df=799)
print(p)
## T-test with different means and variance
stats.ttest_ind_from_stats(x1,s1,n1,x2,s2,n2)


3.5761589403973506
0.00018482198832892038


Ttest_indResult(statistic=5.025702668336442, pvalue=1.1185647741148368e-06)

Since P-value is less than 0.05, Alternative hypothesis is accepted. There is difference between means of two population. There is evidence to conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week.

Problem Statement 12:
Is there evidence to conclude that the number of people preferring Duracell battery is different from the number of people preferring Energizer battery, given the following:
Population 1: Duracell
n1 = 100
x1 = 308
s1 = 84
Population 2: Energizer
n2 = 100
x2 = 254
s2 = 67

In [28]:
n1 = 100
x1 = 308
s1 = 84

n2 = 100
x2 = 254
s2 = 67

The null hypothesis (H0) is no difference between the two populaiton mean i.e x1 = x2

The alternative hypothesis is there is difference between two population i.e. x1 != x2

In [31]:
t_value=(x1-x2)/((s1/(n1**0.5)+(s2)/(n2**0.5)))
print(t_value)
p = 1 - stats.t.cdf(t_value,df=99)
print(p)
## T-test with different means and variance
stats.ttest_ind_from_stats(x1,s1,n1,x2,s2,n2)

3.5761589403973506
0.00027064140928922953


Ttest_indResult(statistic=5.025702668336442, pvalue=1.1185647741148368e-06)

Since P-value is less than 0.05, Alternative hypothesis is accepted. There is difference between means of two population.
there is evidence to conclude that the number of people preferring Duracell battery is different from the number of people preferring Energizer battery. 

Problem Statement 13:
Pooled estimate of the population variance Does the data provide sufficient evidence to conclude that average percentage
increase in the price of sugar differs when it is sold at two different prices?
Population 1: Price of sugar = Rs. 27.50 n1 = 14
x1 = 0.317%
s1 = 0.12%
Population 2: Price of sugar = Rs. 20.00 n2 = 9
x2 = 0.21%
s2 = 0.11%

In [35]:
n1 = 14
x1 = 0.317*27.50
s1 = 0.12*27.50

n2 = 9
x2 = 0.21*20.0
s2 = 0.11*20
df = 14+9-2

The null hypothesis (H0) is no difference between the two different prices i.e x1 = x2

The alternative hypothesis is there is difference between two different prices i.e. x1 != x2

In [41]:
t_value=(x1-x2)/((s1/(n1**0.5)+(s2)/(n2**0.5)))
print(t_value)
p = 1- stats.t.cdf(t_value,df=21)
print(p)
## T-test with different means and variance
stats.ttest_ind_from_stats(x1,s1,n1,x2,s2,n2)

2.7967020221194385
0.005404637398853884


Ttest_indResult(statistic=3.6086382137896145, pvalue=0.0016494680699567412)

Since P-value is less than 0.05, Alternative hypothesis is accepted. There is evidence to conclude that average percentage increase in the price of sugar differs when it is sold at two different prices

Problem Statement 14:
The manufacturers of compact disk players want to test whether a small price reduction is enough to increase sales of their product. Is there evidence that the small price reduction is enough to increase sales of compact disk players?
Population 1: Before reduction
n1 = 15
x1 = Rs. 6598 s1 = Rs. 844
Population 2: After reduction n2 = 12
x2 = RS. 6870
s2 = Rs. 669

In [43]:
n1 = 15
x1 = 6598
s1 =844

n2 = 12
x2 = 6870
s2 = 669
df = 15+12-2

The null hypothesis (H0) is no difference in sale with a small price reduction i.e x1 = x2

The alternative hypothesis is there is difference in sale with a small price reduction i.e. x1 != x2

In [56]:
t_value=(x1-x2)/((s1/(n1**0.5)+(s2)/(n2**0.5)))
print(t_value)
p = 1- stats.t.cdf(t_value,df=25)
print(p)
tcalc= stats.t.ppf(0.05,25)
print(tcalc)

-0.6617304043922669
0.742901180060753
-1.708140761251899


In [52]:
## T-test with different means and variance
stats.ttest_ind_from_stats(x1,s1,n1,x2,s2,n2)

Ttest_indResult(statistic=-0.9098300343990461, pvalue=0.3716010051185177)

Since P-value is more than 0.05, Null hypothesis is accepted. There is evidence that there is no difference in sale with a small price reduction.

Problem Statement 15:
Comparisons of two population proportions when the hypothesized difference is zero Carry out a two-tailed test of the equality of banks’ share of the car loan market in 1980 and 1995.
Population 1: 1980
n1 = 1000
x1 = 53
p1 = 0.53
Population 2: 1985
n2 = 100
x2 = 43
p2= 0.43

The null hypothesis (H0) is no difference between the two populaiton mean i.e x1 - x2=0

The alternative hypothesis is there is difference between two population i.e. x1 -x2 != 0

In [103]:
n1=1000
x1=53
p1=0.53

n2=100
x2=43
p2=0.43

In [108]:
p=(x1+x2)/(n1+n2)
print(p)
##z_value =(p1-p2)/(p*(1-p)*(2/100))**0.5
z_value=1.415
crticalz= 1.645

0.08727272727272728


Conclusion -z_value is in acceptance region. Ho is accpeted. There is no statistically significant difference between banks.

Problem Statement 16:
Carry out a one-tailed test to determine whether the population proportion of traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes are offered as at least 10% higher than the proportion of such buyers when no
sweepstakes are on.
Population 1: With sweepstakes
n1 = 300
x1 = 120
p = 0.40
Population 2: No sweepstakes 
n2 = 700
x2 = 140
p2= 0.20

In [115]:
n1=300
x1=120
p1=0.4

n2=700
x2=140
p2=0.2

D=0.1


The null hypothesis (H0) p1-p2 <0.10

The alternative hypothesis Ha: p1-p2>0.1

In [120]:
z_value =((p1-p2)- D)/((p1*(1-p1)/n1) + (p2*(1-p2)/n2))**0.5
print(z_value)
z = stats.norm.ppf(0.001,0,1)
print(z)

3.1180478223116177
-3.090232306167813


Conclusion - z_value is in rejection region. Ho is rejected at common level of siginificance

Problem Statement 17: 
A die is thrown 132 times with the following results: Number turned up: 1, 2, 3, 4, 5, 6
Frequency: 16, 20, 25, 14, 29, 28
Is the die unbiased? Consider the degrees of freedom as 𝑝 −1

Ho = Die is not biased
H1= Die is biased

In [138]:
obv = [16,20,25,14,29,28]
df_obv=pd.DataFrame(obv, columns=['ob'])
df_obv['exp'] = [22,22,22,22,22,22]
df_obv['ch']= (df_obv['ob'] - df_obv['exp'])**2/df_obv['exp']
chi_square = df_obv['ch'].sum()
print(f"chi_square x^2: {chi_square}")
df = 6-1
ch_critical=stats.chi2.ppf(0.95,5)
print(ch_critical)

chi_square x^2: 9.0
11.070497693516351


Conclusion - Calculated chi_square value is in acceptance region. Ho is accepted, therefore die is unbaised.

Problem Statement 18: 
In a certain town, there are about one million eligible voters. A simple random sample of 10,000 eligible voters was chosen to study the relationship between gender and participation in the last election. The results are summarized in the
following 2X2 (read two by two) contingency table:
oberved Table  Men Women Total
        Voted 2792 3591 6386
 Didn't Voted 1486 2131 3617
        Total 4278 5722
Expected Table Men Women
        Voted 2731 3652
  Didn't Vote 1547 2070
We would want to check whether being a man or a woman (columns) is independent of having voted in the last election (rows). In other words, is “gender and voting independent”?


In [146]:
stats.chisquare([2792, 3591, 1486, 2131], f_exp=[2731, 3652, 1547, 2070])

Power_divergenceResult(statistic=6.584283456757529, pvalue=0.08639718274463479)

Ho = gender is independent of voting
H1= gender is not independent of voting

In [ ]:
df = (2-1)*(2-1) # two row n columns

In [162]:
ch_critical1=stats.chi2.ppf(0.95, 1)
print(ch_critical1)

3.841458820694124


In [ ]:
Conclusion - Calculated chi_square value( 6.5) is in critical region. Ho is rejected, therefore voting pattern and gender status are not statistically dependent on one another.

Problem Statement 19:
A sample of 100 voters are asked which of four candidates they would vote for in an election. The number supporting each candidate is given below:
           Higgins Reardon White charlton 
   observed    41  19      24    16
   Expected    25  25      25    25

Do the data suggest that all candidates are equally popular? [Chi-Square = 14.96, with 3 df, 𝑝 0.05 .


Ho - Candidates are equally popuplar
H1- Candidates are not equally popuplar

In [192]:
chi_calc = stats.chisquare([41, 19, 24, 16], f_exp=[25, 25, 25, 25])
print(chi_calc)

Power_divergenceResult(statistic=14.959999999999999, pvalue=0.001851152735284203)


Chi-Square_critical (0.05,3) = 7.82

Conclusion - Calculated chi_square value( 14.96) is in critical region. Ho is rejected therefore candidates are not equally popuplar.

Problem Statement 20:
Children of three ages are asked to indicate their preference for three photographs of
adults. Do the data suggest that there is a significant relationship between age and
photograph preference? What is wrong with this study? [Chi-Square = 29.6, with 4 df: 𝑝 < 0.05].

observed 	A 	B	 C		Expected A 	B	 C
    5-6yrs	18	22	20	60	5-6yrs	12	18	30
    7-8yrs	2	28	40	70	7-8yrs	14	21	35
    9-10yrs	20	10	40	70	9-10yrs	14	21	35
            40	60	100					
            Total	200					


Ho - There is a significant relationship between age and photograph preference
H1- There is a no significant relationship between age and photograph preference

In [193]:
chi_calc = stats.chisquare([18, 22, 20, 2,28,40,20,10,40], f_exp=[12, 18, 30, 14,21,35,14,21,35])
print(chi_calc)

Power_divergenceResult(statistic=29.603174603174608, pvalue=0.00024837483246733366)


In [ ]:
df=(4-1)*(3-1)
print(df)
Chi-Square_critical (0.05,6) = 12.59

In [198]:
ch_critical1=stats.chi2.ppf(0.95, 6)
print(ch_critical1)

12.591587243743977


Conclusion - Calculated chi_square value( 29.60) falls in critical region. Ho is rejected,There is a no significant relationship between age and photograph preference.

Problem Statement 21:
A study of conformity using the Asch paradigm involved two conditions: one where one confederate supported the true judgement and another where no confederate gave the correct response.
Is there a significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df: 𝑝 < 0.05].


Observed 	Support	No Support		 Expected Support	No Support
   Conform	18	    22	   		  Conform	    11	     29
Not Conform	2	   28	   	     Not Conform	9	     21
	Total	70					


Ho - There is a significant relationship between "support" and "no support" conditions
H1 - There is a no significant relationship between "support" and "no support" conditions

In [194]:
chi_calc = stats.chisquare([18, 22, 2,28], f_exp=[11, 29, 9, 21])
print(chi_calc)

Power_divergenceResult(statistic=13.921978404737025, pvalue=0.003013300191723246)


In [ ]:
df=(2-1)*(2-1)
print(df)
Chi-Square_critical (0.05,1) = 3.84

In [199]:
ch_critical1=stats.chi2.ppf(0.95, 1)
print(ch_critical1)

3.841458820694124


Conclusion - Calculated chi_square value( 13.92) falls in critical region. Ho is rejected,There is a no significant relationship between "support" and "no support" conditions.

Problem Statement 22: 
We want to test whether short people differ with respect to their leadership qualities (Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget MP's are there?) The following table shows the frequencies with which 43 short people and 52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a
relationship between height and leadership qualities?
[Chi-Square = 10.71, with 2 df: 𝑝 < 0.01]

observed 	short	Tall			Expected	short	Tall
Leader	    12	    32		44	    Leader	      20	   24
Follower	22	    14		36	    Follower	  16	    20
Unclassifiable	9	6		15	    Unclassifiable	7	8
Total 95

Ho - There is a significant relationship between "height" and leadership
H1 - There is a no significant relationship between "height" and leadership

In [196]:
chi_calc = stats.chisquare([12, 32, 22,14,9,6], f_exp=[20, 24,16,20,7,8])
print(chi_calc)

Power_divergenceResult(statistic=10.988095238095239, pvalue=0.05161653140950149)


In [ ]:
df=(3-1)*(3-1)
print(df)
Chi-Square_critical (0.05,4) = 9.49

In [200]:
ch_critical1=stats.chi2.ppf(0.95, 4)
print(ch_critical1)

9.487729036781154


Conclusion - Calculated chi_square value( 10.98) falls in critical region. Ho is rejected,There is a no significant relationship between "height" and leadership.

Problem Statement 22:
Each respondent in the Current Population Survey of March 1993 was classified as employed, unemployed, or outside the labor force. The results for men in California age 35-44 can be cross-tabulated by marital status, as follows:
Men of different marital status seem to have different distributions of labor force status. Or is this just chance variation? (you may assume the table results from a simple random sample.)

Observed 	Married	separated/widowed	Never Married		Expected	Married	separated/widowed	Never Married
Employed	679	         103	        114	                 Employed	654	    109	        133
Unemployed	63	         10	            20	                Unemployed	68	    11	   14
Not in lobor Force	42	 18	            25	         Not in lobor Force	62	    11	     13
	784	131	159					
Total	1074					


In [197]:
chi_calc = stats.chisquare([679, 103, 114,63,10,20,42,18,25], f_exp=[654, 109,133,68,11,14,62,11,13])
print(chi_calc)

Power_divergenceResult(statistic=29.013284591853783, pvalue=0.0003154003990461922)


In [ ]:
df=(3-1)*(3-1)
print(df)
Chi-Square_critical (0.05,4) = 9.49
pvalue=0.0003 < 0.05

In [202]:
ch_critical1=stats.chi2.ppf(0.95, 4)
print(ch_critical1)

9.487729036781154


In [ ]:
Conclusion - Calculated chi_square value( 29.01) falls in critical region. Ho is rejected,There is a no relationship between rows and columns.